GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -801,12 +801,13 @@ protected override void DisposeCustomResources()
 
         internal sealed class BranchComparer : IComparer<string>
         {
-            private const string RemoteBranchPrefix = "remotes/";
-
-            private static readonly Regex _importantRepoRegex = new($"^{RemoteBranchPrefix}(origin|upstream)/",
-                RegexOptions.Compiled | RegexOptions.CultureInvariant);
-            private static readonly Regex _remoteMasterRegex = new($"^{RemoteBranchPrefix}.*/(main|master)[^/]*$",
-                RegexOptions.Compiled | RegexOptions.CultureInvariant);
+            private const string _remoteBranchPrefix = "remotes/";
+            private static readonly Regex _importantRepoRegex = new($"^{_remoteBranchPrefix}({AppSettings.PrioritizedRemoteNames.Trim().Replace(';', '|')})/",
+                RegexOptions.Compiled | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant);
+            private static readonly Regex _remoteImportantBranchRegex = new($"^{_remoteBranchPrefix}[^/]+/({AppSettings.PrioritizedBranchNames.Trim().Replace(';', '|')})$",
+                RegexOptions.Compiled | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant);
+            private static readonly Regex _localImportantBranchRegex = new($"^({AppSettings.PrioritizedBranchNames.Trim().Replace(';', '|')})$",
+                RegexOptions.Compiled | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant);
```

Comment:
Modified settings would require a restart in order to come into effect.

Reply:
Add to tooltip?

Reply:
> Add to tooltip?

🤔 Would be imperfect, because it makes my day when I write a correct regex in the first attempt. On the other hand, no complex regexes are expected here.
Users will wonder why it takes effect for the side panel and not in the commit info.